src/client/desktop_gui/lib/grpc_client.dart

```diff
@@ -0,0 +1,199 @@
+    return _client.launch(Stream.value(request), options: CallOptions());
+  }
+
+  Future<StartReply?> start(Iterable<String> names) async {
+    var request = StartRequest(
+      instanceNames: InstanceNames(instanceName: names),
+    );
+    return _client.start(Stream.value(request)).firstOrNull;
+  }
+
+  Future<StopReply?> stop(Iterable<String> names) async {
+    var request = StopRequest(
+      instanceNames: InstanceNames(instanceName: names),
+    );
+    return _client.stop(Stream.value(request)).firstOrNull;
+  }
+
+  Future<SuspendReply?> suspend(Iterable<String> names) async {
+    var request = SuspendRequest(
+      instanceNames: InstanceNames(instanceName: names),
+    );
+    return _client.suspend(Stream.value(request)).firstOrNull;
+  }
+
+  Future<RestartReply?> restart(Iterable<String> names) async {
+    var request = RestartRequest(
+      instanceNames: InstanceNames(instanceName: names),
+    );
+    return _client.restart(Stream.value(request)).firstOrNull;
+  }
+
+  Future<DeleteReply?> delete(Iterable<String> names) async {
+    var request = DeleteRequest(
+      instanceNames: InstanceNames(instanceName: names),
+    );
+    return _client.delet(Stream.value(request)).firstOrNull;
+  }
+
+  Future<RecoverReply?> recover(Iterable<String> names) async {
+    var request = RecoverRequest(
+      instanceNames: InstanceNames(instanceName: names),
+    );
+    return _client.recover(Stream.value(request)).firstOrNull;
+  }
+
+  Future<DeleteReply?> purge(Iterable<String> names) async {
+    var request = DeleteRequest(
+      instanceNames: InstanceNames(instanceName: names),
+      purge: true,
+    );
+    return _client.delet(Stream.value(request)).firstOrNull;
+  }
+
+  Stream<List<VmInfo>> infoStream() async* {
+    Object? lastError;
+    await for (final _ in Stream.periodic(1.seconds)) {
+      try {
+        final reply = await _client.info(Stream.value(InfoRequest())).last;
+        yield reply.info;
+        lastError = null;
+      } catch (error, stackTrace) {
+        if (error != lastError) yield* Stream.error(error, stackTrace);
+        lastError = error;
+      }
+    }
+  }
+
+  Future<List<FindReply_ImageInfo>> find() {
+    return _client
+        .find(Stream.value(FindRequest(showBlueprints: true, showImages: true)))
+        .single
+        .then((r) => r.imagesInfo);
+  }
+
+  Future<String> version() {
+    return _client
+        .version(Stream.value(VersionRequest()))
+        .single
+        .then((reply) => reply.version);
+  }
+}
+
+class CustomChannelCredentials extends ChannelCredentials {
+  final List<int> certificateChain;
+  final List<int> certificateKey;
+
+  CustomChannelCredentials({
+    required List<int> certificate,
+    required this.certificateKey,
+    String? authority,
+  })  : certificateChain = certificate,
+        super.secure(
+            certificates: certificate,
+            authority: authority,
+            onBadCertificate: allowBadCertificates);
```

Comment:
I'm not really sure why, but our certificates fail validation, and the Dart gRPC client doesn't want to use them. This is a way of telling it to use them anyway, since the daemon knows how to deal with them. I'm not really sure why they fail...

Reply:
Hmm, well I think we should figure out why exactly it is that the Dart gRPC thinks they are bad. Once we know that, we can decide whether it is something we can safely ignore, easily fix, or something else.

Reply:
OK, Chris just dug a little and found that "bad" in this context just means unverifiable with a certificate authority. We don't really care for that because this is for ourselves, so we're good here. I would only ask for a comment explaining that.